### DISCUSSION TITLE
why client connected return tiwce different sid.

### DISCUSSION THREAD

**Discussion by nadjin** [socketio/socket.io#4528] [Q&A]
**Description**

the sever socket.io is 4.x.

ver: 2.x
![image](https://user-images.githubusercontent.com/23130767/202970681-2723c8fd-9bb1-4b9d-8d77-fc598ff3a46b.png)

ver: 4.x
![image](https://user-images.githubusercontent.com/23130767/202970570-462c24aa-bf24-4f24-8343-2cb0259be454.png)

**Question**

why 4.x client conneted return twice different sid, I use the second sid can receive message?

**code**

> client

```js
this.socket = io.connect(this.host, {
    transports:['websocket']
    // autoConnect:false
});
```

> server 4.x

nestjs

```typescript

@WebSocketGateway( { transports: ['websocket'], pingTimeout: 60000, pingInterval: 25000, allowEIO3: true })
export class HallGateway implements NestGateway {
  @WebSocketServer()
  private readonly server: Server;
  private readonly logger = new Logger(HallGateway.name);

  handleConnection(client: Socket, ...args: any[]): any {
    this.logger.log('socket id is: ', client.id);
  }

  handleDisconnect(client: Socket): any {
    this.logger.log(JSON.stringify(client.rooms));
  }
}
```

**Comment by darrachequesne** [comment]
Hi! There are indeed two different session ID:

- one at the Engine.IO level, so the server knows the identity of the client across HTTP requests
- one at the Socket.IO level, which can be used for private messages

Reference: https://socket.io/docs/v4/how-it-works/
